.github/workflows/build-java.yaml

```diff
@@ -0,0 +1,109 @@
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-java
+  cancel-in-progress: true
+
+jobs:
+  build-jni:
+    strategy:
+      matrix:
+        # start with the most popular ones
+        include:
+          - os: macos-latest
+            target: x86_64-apple-darwin
+            suffix: ".dylib"
+            jdk: 8
+          - os: macos-latest
+            target: aarch64-apple-darwin
+            suffix: ".dylib"
+            # https://adoptium.net/supported-platforms/
+            jdk: 11
+          - os: ubuntu-latest
+            target: x86_64-unknown-linux-gnu
+            suffix: ".so"
+            jdk: 8
+          - os: windows-latest
+            target: x86_64-pc-windows-msvc
+            suffix: ".dll"
+            jdk: 8
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Install JDK 8
+        uses: actions/setup-java@v3
+        with:
+          distribution: "temurin"
+          java-version: ${{ matrix.jdk }}
+      - name: Install targets
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          override: true
+          toolchain: stable
+          target: ${{ matrix.target }}
+      - name: Build
+        #uses: richb-hanover/cargo@v1.1.0
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --target=${{ matrix.target }} --release -p prql-java
+      - uses: actions/upload-artifact@v3
+        with:
+          name: ${{ matrix.target }}
+          path: target/${{ matrix.target }}/release/*prql_java${{ matrix.suffix }}
+          if-no-files-found: error
+
+  build-java:
+    needs:
+      - build-jni
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install JDK 8 and Maven
+        uses: actions/setup-java@v3
+        with:
+          distribution: "temurin"
+          java-version: 8
+          cache: "maven"
+      - uses: actions/download-artifact@v3
+        with:
+          path: prql-java/prql-api/src/main/resources/META-INF/native
+      - name: Build and install libraries
+        working-directory: prql-java
+        run:
+          mvn --batch-mode --show-version --strict-checksums
+          -Dmaven.wagon.rto=30000 -DskipTests package
+      - uses: actions/upload-artifact@v3
+        with:
+          name: prql-jdbc
+          path: prql-java/prql-jdbc/target/*.jar
+          if-no-files-found: error
+      # - name: Release Maven package
+      #   uses: samuelmeuli/action-maven-publish@v1
+      #   with:
+      #     directory: prql-java
+      #     maven_profiles: release
+      #     maven_args:
+      #       -q --batch-mode --show-version --strict-checksums --threads C1
+      #       -Dmaven.wagon.rto=30000
+      #     gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
+      #     gpg_passphrase: ${{ secrets.GPG_PASSPHRASE }}
+      #     nexus_username: ${{ secrets.SONATYPE_USER }}
```

Comment:
I looked into getting a username & password. But:
- I see that the action has been archived. Are there any alternatives?
- Are you familiar with how to get a Sonatype login? I can look more if not, but I couldn't immediately find how to get one.